server/tests/post_incomplete_json_body/__init__.py

```diff
@@ -0,0 +1,34 @@
+import json
+import pytest
+
+class TestPostIncompleteJSONBody:
+    # pylint: disable=no-self-use
+    @pytest.fixture(scope='class')
+    def removed_field(self, antibody_incomplete_data):
+        return antibody_incomplete_data[1]
+
+    @pytest.fixture(scope='class')
+    def incomplete_data(self, antibody_incomplete_data):
+        return antibody_incomplete_data[0]
+
+    @pytest.fixture(scope='class')
+    def response(self, client, headers, incomplete_data):
+        return client.post(
+            '/antibodies', data=json.dumps(incomplete_data), headers=headers
+        )
+
+    def test_post_with_incomplete_json_body_should_return_406(
+            self, response
+        ):
+        """POST /antibodies with incomplete JSON body
+         should return 406 NOT ACCEPTABLE"""
+        assert response.status == '406 NOT ACCEPTABLE'
```

Comment:
Should this be a 400 error instead of using 406? A 406 status code is normally matching the following headers from request:
```
Accept
Accept-Encoding
Accept-Language
```